### DISCUSSION TITLE
Are repositories IDs assigned in chronological order?

### DISCUSSION THREAD

**Discussion by agilot** [community/community#142150] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hi,
I am doing a code study and using GitHub REST API for that purpose.
From what I have seen so far repositories IDs seem to be chronologically assigned. Is that the case? If yes, is there a reliable source that confirms it?

Thank you!

**Comment by jheisonnovak** [comment, marked as answer]
Yes, repository IDs in the GitHub REST API generally increase as new repositories are created, giving the impression of being chronological. However, GitHub does not officially guarantee that these IDs are strictly sequential or always reflect the exact order of creation. Factors like database replication could cause slight variations. For more details, check the [GitHub REST API documentation](https://docs.github.com/en/rest), but keep in mind that ID generation is considered an internal detail that could change.
